### PR TITLE
wearable trashbags and chicken bucket hats & storage item tweak

### PR DESCRIPTION
**Pull request by RealestEstate**
lets you wear an empty trashbag on the outer suit slot and an empty kfc bucket as a hat. also tweaks how the code handles wearing a storage on item on a slot and having it not function as a storage item while in that slot (ie wearing plastic bags on your head).

sprites courtesy of Blithering, aka the Grand Duke of the Trashin' Fashion Scene.

:cl:
 * rscadd: Empty trash bags can be worn on the suit slot, and empty chicken buckets can be worn as hats. Sprites courtesy of Blithering

### CODE REVIEW

**Comment by SonixApache** [comment]
unnecessary tweaks
bring back hat storage

File: code/game/objects/items/weapons/storage/storage.dm [hunk 1]
```diff
@@ -665,3 +672,10 @@
 /obj/item/weapon/storage/proc/mass_remove(var/atom/A)
 	for(var/obj/item/O in contents)
 		remove_from_storage(O, A)
+
+/obj/item/weapon/storage/mob_can_equip(mob/M, slot, disable_warning = 0, automatic = 0)
+	//Forbids wearing a storage item in a  no_storage_slot (ie plastic bags over head) with something already inside
+	.=..()
+	for (var/i in no_storage_slot)
```
Comment (unid15) [inline code/game/objects/items/weapons/storage/storage.dm:679]:
```
if(no_storage_slot.Find(slot) && contents.len)
    return CANNOT_EQUIP
```

File: code/game/objects/items/weapons/storage/storage.dm [hunk 2]
```diff
@@ -29,6 +29,8 @@
 	var/foldable = null	// BubbleWrap - if set, can be folded (when empty) into a sheet of cardboard
 	var/foldable_amount = 1 // Number of foldables to produce, if any - N3X
 	var/internal_store = 0
+	var/list/no_storage_slot = new/list()//if the item is equipped in a slot that is contained in this list, the item will act purely as a clothing item and not a storage item (ie plastic bags over head)
```
Comment (PJB3005) [inline code/game/objects/items/weapons/storage/storage.dm:32]: Use `list()`, not `new/list()`

File: code/game/objects/items/weapons/storage/storage.dm [hunk 3]
```diff
@@ -213,9 +215,14 @@
 
 //This proc return 1 if the item can be picked up and 0 if it can't.
 //Set the stop_messages to stop it from printing messages
-/obj/item/weapon/storage/proc/can_be_inserted(obj/item/W as obj, stop_messages = 0)
+/obj/item/weapon/storage/proc/can_be_inserted(obj/item/W as obj, stop_messages = 0,mob/M, slot)
```
Comment (PJB3005) [inline code/game/objects/items/weapons/storage/storage.dm:218]: Where are you changing the usage of this proc to pass the new arguments?

**Review by PJB3005 — CHANGES_REQUESTED** [review]

**Comment by RealestEstate** [comment]
I think I've addressed the changes as they were suggested

**Comment by Lokiamis** [comment]
My favourite pr

**Comment by RealestEstate** [comment]
why is icons/mob/head.dmi a conflicting file? is there another PR that is using it?

**Comment by Lokiamis** [comment]
Because of #17240

**Review by PJB3005 — APPROVED** [review]

**Comment by SkowronX** [comment]
da trashman

**Comment by DamianX** [comment]
please fix the conflicts

**Pull request closed without merging**